__tests__/integration.js

```diff
@@ -82,6 +84,29 @@ test('--mutex network', async () => {
   ]);
 });
 
+test('--mutex network 2', async () => {
+  const cwd = await makeTemp();
+  await fs.writeFile(path.join(cwd, '.yarnrc'), '--mutex network\n');
+
+  const promises = [];
+
+  for (let t = 0; t < 40; ++t) {
+    const subCwd = path.join(cwd, String(t));
+
+    await fs.mkdirp(subCwd);
+    await fs.writeFile(
+      path.join(subCwd, 'package.json'),
+      JSON.stringify({
+        scripts: {test: 'node -e "setTimeout(function(){}, process.argv[1])"'},
+      }),
+    );
+
+    promises.push(runYarn(['run', 'test', '100'], {cwd: subCwd}));
```

Comment:
This made me realize that we should probably limit `mutex` to places where we touch cache. It doesn't make much sense to block `yarn run` calls with a mutex, right? (future breaking TODO)

Reply:
Yep, I also witnessed this error first-hand last week, an internal build was failing exactly because of this. I solved it temporarily by restricting the option range to `install`, `add`, `remove`, and `upgrade`, but a more robust solution would to give commands the ability to export a variable when they need to be "mutex-aware", kinda like what we're doing with [`noArgument`](https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/import.js#L33).